src/main/groovy/edu/oregonstate/mist/locations/health/LocationHealthCheck.groovy

```diff
@@ -0,0 +1,37 @@
+package edu.oregonstate.mist.locations.health
+
+import com.codahale.metrics.health.HealthCheck
+import com.codahale.metrics.health.HealthCheck.Result
+
+class LocationHealthCheck extends HealthCheck {
+    final Map<String, String> locationConfiguration
+
+    LocationHealthCheck(Map<String, String> locationConfiguration) {
+        this.locationConfiguration = locationConfiguration
+    }
+
+    protected Result check() {
+        Result.unhealthy("default health check needs to be overwritten")
+    }
+
+    /**
+     * Verify that a url is accessible by DW and doesn't return an
+     * empty string.
+     *
+     * @param url
+     * @return
+     */
+    protected Result checkUrl(String url) {
+        try {
+            String urlText = new URL(url).text
+
+            if (urlText) {
+                Result.healthy()
+            } else {
+                Result.unhealthy("url (${url}) was empty or null")
+            }
```

Comment:
Hmm, testing !url just validates that the url provided is not empty. I want the healthcheck to verify that there's some data coming back from the url configured in the datasources. I'm going to modify the unhealthy message to make things more clear.